### PR TITLE
feat: update to wagmi 0.9

### PR DESCRIPTION
**Pull request by jxom**
Upgrading to wagmi 0.9!

### CODE REVIEW

**Comment by armatrix** [comment]
plz update to wagmi 0.9

**Comment by davidthomasparks** [comment]
Currently blocked by this PR. The installation docs have you installing the latest 0.9 versions but there are huge compat issues.

**Comment by tuler** [comment]
wagmi 0.9.1 is out.

ps: why `pnpm-lock.yaml` is not committed?

**Comment by SimonHausdorf** [comment]
wagmi 0.9.2 got released. Can we have this merged soon as some important features are missing now and blocking us from using it.

**Review by nickbytes — APPROVED** [review]

File: packages/rainbowkit/src/hooks/useMainnet.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
-import { Chain, useProvider, chain as wagmiChains } from 'wagmi';
+import { Chain, useProvider } from 'wagmi';
+import { mainnet } from 'wagmi/chains';
 
 export function useMainnet() {
```
Comment (nickbytes) [inline packages/rainbowkit/src/hooks/useMainnet.ts:4]: Is linting failing for you at `useProvider` @jxom ?

**Comment by armatrix** [comment]
Thanks guys！

**Pull request merged**